src/main/java/PositionAtTime.java

```diff
@@ -0,0 +1,85 @@
+import frame.StoredProcs;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class PositionAtTime
+{
+    StoredProcs sp;
+    String[] currencies;
+    double initialUSD;
+
+    public PositionAtTime(StoredProcs sp, String[] currencies, double initialUSD)
+    {
+        this.sp = sp;
+        this.currencies = currencies;
+        this.initialUSD = initialUSD;
+    }
+
+    // todo: return map of currency type and amount current held in that currency
+    public HashMap<String, Double> getPositionAtTime(LocalDate date) throws SQLException
+    {
+        HashMap<String, Double> positions = new HashMap<>();
+        positions.put("USD", initialUSD); // adding separately because not included in currency list
+        for (String curr : currencies)
+        {
+            positions.put(curr, 0.0);
+        }
+
+        ArrayList<String[]> records = sp.getAllTransactions();
+        // form: date, vendor, quantity, type, currency, rate, maturity
+        // quantity: amount in FX
+        // rate: FX per dollar
+        if (records != null)
+        {
+            for (String[] record : records)
+            {
+                String t_date = record[0];
+                if (LocalDate.parse(t_date).isAfter(date))
+                {
+                    continue;
+                }
+
+                // otherwise:
+                String type = record[3];
+                String maturityD = record[6];
+                LocalDate maturityLD = maturityD == null || maturityD.equals("") ?
+                        null :
+                        LocalDate.parse(maturityD);
+
+                boolean isSpot = type.equals("spot");
+                boolean isGoodFuture = type.equals("future")
```

Comment:
if we get around to it, maybe we can add a display of pending transactions/positions for the forwards that havent matured yet.